docs/src/tutorial.md

```diff
@@ -0,0 +1,157 @@
+If you want to use different values from the default ones, you can pass the parameters as keyword arguments to the algorithm constructor. For example, you can limit the depth of the search tree to `6` the following way
+
+```@example tutorial
+enclose(f, X, BranchAndBoundEnclosure(; maxdepth=6))
+```
+
+Generally, tuning parameters can be a good idea to achieve the desired accuracy-tradeoff in your application.
+
+### Combining different methods
+
+Sometimes there is no "best" algorithm, one might give a tighter estimate of the range upper bound and another a tighter estimate on the lower bound. In this case, the result can
+be achieved by taking the intersection of both results. For example, let us consider the function ``g(x) = x^2 - 2x + 1``, which we want to evaluate over the domain ``X_g = [0, 4]``. Let us first try using plain interval arithmetic
+
+```@example tutorial
+g(x) = x^2 - 2*x + 1
+Xg = 0..4
+enclose(g, Xg, NaturalEnclosure()) # this is equivalent to enclose(g, Xg)
+```
+
+Now let us bound the range using [`MeanValueEnclosure`](@ref), which uses the mean-value form of the function.
+
+```@example tutorial
+enclose(g, Xg, MeanValueEnclosure())
+```
+
+As you can see, there is no clear winner and a better enclosure could be obtained by taking the intersection of the two results. This can be easily done in one pass by passing to `enclose` a vector of methods
+
+```@example tutorial
+enclose(g, Xg, [NaturalEnclosure(), MeanValueEnclosure()])
+```
+
+### Using algorithms from external libraries
+
+Not all algorithms are implemented in this package, some are from external libraries. To keep the start-up time of RangeEnclosures.jl reasonable, routines from external packages
+are not imported by default and if you want to use those, you will need to also import the required package. For example, suppose you want to bound the previous function using the Moore-Skelboe algorithm. Trying the following will fail
+
+```julia
+julia> enclose(f, X, MooreSkelboeEnclosure())
+```
+
+```
+ERROR: AssertionError: package 'IntervalOptimisation' not loaded (it is required for executing `enclose`)
+...
+```
+
+This is because the algorithm is implemented in [`IntervalOptimisation.jl`](https://github.com/juliaintervals/intervaloptimisation.jl) and to use it you need to import it first (note that you need to have it installed before importing it). Let us fix our example
+
+```@example tutorial
+using IntervalOptimisation
+enclose(f, X, MooreSkelboeEnclosure())
+```
+
+### Bounding multivariate functions
+
+While our previous examples were in 1D, the techniques generalize to multivariate functions ``\mathbb{R}^n\rightarrow\mathbb{R}``, the only difference is that the domain, instead of being an interval, should be an `IntervalBox`. For example, if we want to bound the function
```

Comment:
```suggestion
While our previous examples were in one dimension, the techniques generalize to multivariate functions ``\mathbb{R}^n\rightarrow\mathbb{R}``, the only difference is that the domain, instead of being an interval, should be an `IntervalBox`. For example, if we want to bound the function
```